build error when delta OTA feature is enabled

**Describe the bug**
When CONFIG_ENABLE_DELTA_OTA is enabled, build failed with error of undefined reference.
However, esp_delta_ota module exists in my project.

Error log as below : 
/home/zpc/.espressif/tools/riscv32-esp-elf/esp-14.2.0_20251107/riscv32-esp-elf/bin/../lib/gcc/riscv32-esp-elf/14.2.0/../../../../riscv32-esp-elf/bin/ld: esp-idf/chip/lib/libCHIP.a(ESP32.OTAImageProcessorImpl.cpp.o): in function `chip::OTAImageProcessorImpl::HandlePrepareDownload(int)':
/IDF_BUILD/esp-idf/chip/../../../../../../../opt/espressif/esp-matter/connectedhomeip/connectedhomeip/config/esp32/third_party/connectedhomeip/src/platform/ESP32/OTAImageProcessorImpl.cpp:348:(.text._ZN4chip21OTAImageProcessorImpl21HandlePrepareDownloadEi+0x98): undefined reference to `esp_delta_ota_init'
/home/zpc/.espressif/tools/riscv32-esp-elf/esp-14.2.0_20251107/riscv32-esp-elf/bin/../lib/gcc/riscv32-esp-elf/14.2.0/../../../../riscv32-esp-elf/bin/ld: esp-idf/chip/lib/libCHIP.a(ESP32.OTAImageProcessorImpl.cpp.o): in function `chip::OTAImageProcessorImpl::HandleFinalize(int)':
/IDF_BUILD/esp-idf/chip/../../../../../../../opt/espressif/esp-matter/connectedhomeip/connectedhomeip/config/esp32/third_party/connectedhomeip/src/platform/ESP32/OTAImageProcessorImpl.cpp:389:(.text._ZN4chip21OTAImageProcessorImpl14HandleFinalizeEi+0x2a): undefined reference to `esp_delta_ota_finalize'
/home/zpc/.espressif/tools/riscv32-esp-elf/esp-14.2.0_20251107/riscv32-esp-elf/bin/../lib/gcc/riscv32-esp-elf/14.2.0/../../../../riscv32-esp-elf/bin/ld: /IDF_BUILD/esp-idf/chip/../../../../../../../opt/espressif/esp-matter/connectedhomeip/connectedhomeip/config/esp32/third_party/connectedhomeip/src/platform/ESP32/OTAImageProcessorImpl.cpp:408:(.text._ZN4chip21OTAImageProcessorImpl14HandleFinalizeEi+0x7c): undefined reference to `esp_delta_ota_deinit'
/home/zpc/.espressif/tools/riscv32-esp-elf/esp-14.2.0_20251107/riscv32-esp-elf/bin/../lib/gcc/riscv32-esp-elf/14.2.0/../../../../riscv32-esp-elf/bin/ld: esp-idf/chip/lib/libCHIP.a(ESP32.OTAImageProcessorImpl.cpp.o): in function `chip::OTAImageProcessorImpl::HandleProcessBlock(int)':
/IDF_BUILD/esp-idf/chip/../../../../../../../opt/espressif/esp-matter/connectedhomeip/connectedhomeip/config/esp32/third_party/connectedhomeip/src/platform/ESP32/OTAImageProcessorImpl.cpp:535:(.text._ZN4chip21OTAImageProcessorImpl18HandleProcessBlockEi+0xf6): undefined reference to `esp_delta_ota_feed_patch'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
ninja failed with exit code 1, output of the command is in the /home/zpc/projects/matter-fw-cn-room-controller/build/log/idf_py_stderr_output_1410335 and /home/zpc/projects/matter-fw-cn-room-controller/build/log/idf_py_stdout_output_1410335

**Environment**
- ESP-Matter Commit Id: 30b0f6e
- ESP-IDF Commit Id: v5.5.3
- SoC (eg: ESP32 or ESP32-C3): esp32c5
- Device Logs (Please attach the log file):

[idf_py_stderr_output_1410335.txt](https://github.com/user-attachments/files/26991795/idf_py_stderr_output_1410335.txt)
[idf_py_stdout_output_1410335.txt](https://github.com/user-attachments/files/26991794/idf_py_stdout_output_1410335.txt)

- Host Machine OS: ubuntu 24.04 LTS
- Host Machine Python version: 3.12.3

- Commissioner app and versions if present:
- Home hub app and versions if present:
- Commissioner's logs if present:

**Any additional details**
...

--- a/config/esp32/components/chip/CMakeLists.txt
+++ b/config/esp32/components/chip/CMakeLists.txt
@@ -487,6 +487,10 @@ if (CONFIG_ENABLE_ENCRYPTED_OTA)
     list(APPEND matter_requires espressif__esp_encrypted_img)
 endif()
 
+if (CONFIG_ENABLE_DELTA_OTA)
+    list(APPEND matter_requires espressif__esp_delta_ota)
+endif()
+
 add_prebuilt_library(matterlib "${CMAKE_CURRENT_BINARY_DIR}/lib/libCHIP.a"
                      REQUIRES ${matter_requires})
 

--- a/src/platform/ESP32/OTAImageProcessorImpl.cpp
+++ b/src/platform/ESP32/OTAImageProcessorImpl.cpp
@@ -261,6 +261,7 @@ esp_err_t OTAImageProcessorImpl::DeltaOTAWriteCallback(const uint8_t * buf, size
             esp_image_header_t * header = (esp_image_header_t *) headerData;
             if (!VerifyChipId(header->chip_id))
             {
+                headerDataRead = 0;
                 return ESP_ERR_INVALID_VERSION;
             }
             imageProcessor->chipIdVerified = true;

--- a/src/platform/ESP32/OTAImageProcessorImpl.cpp
+++ b/src/platform/ESP32/OTAImageProcessorImpl.cpp
@@ -322,7 +322,7 @@ void OTAImageProcessorImpl::HandlePrepareDownload(intptr_t context)
     if (imageProcessor->mDeltaOTAUpdateHandle == NULL)
     {
         ChipLogError(SoftwareUpdate, "esp_delta_ota_init failed");
-        imageProcessor->mDownloader->OnPreparedForDownload(CHIP_ERROR_INTERNAL);
+        LogErrorOnFailure(imageProcessor->mDownloader->OnPreparedForDownload(CHIP_ERROR_INTERNAL));
         return;
     }
 #endif // CONFIG_ENABLE_DELTA_OTA
@@ -332,7 +332,7 @@ void OTAImageProcessorImpl::HandlePrepareDownload(intptr_t context)
     if (chipError != CHIP_NO_ERROR)
     {
         ChipLogError(SoftwareUpdate, "Failed to start decryption process, err:%" CHIP_ERROR_FORMAT, chipError.Format());
-        imageProcessor->mDownloader->OnPreparedForDownload(chipError);
+        LogErrorOnFailure(imageProcessor->mDownloader->OnPreparedForDownload(chipError));
         return;
     }
 #endif // CONFIG_ENABLE_ENCRYPTED_OTA
@@ -360,7 +360,7 @@ void OTAImageProcessorImpl::HandleFinalize(intptr_t context)
     {
         ChipLogError(SoftwareUpdate, "Failed to end pre encrypted OTA");
         esp_ota_abort(imageProcessor->mOTAUpdateHandle);
-        imageProcessor->ReleaseBlock();
+        LogErrorOnFailure(imageProcessor->ReleaseBlock());
         PostOTAStateChangeEvent(DeviceLayer::kOtaDownloadFailed);
         return;
     }
@@ -372,7 +372,7 @@ void OTAImageProcessorImpl::HandleFinalize(intptr_t context)
     {
         ESP_LOGE(TAG, "esp_delta_ota_finalize() failed (%s)!", esp_err_to_name(err));
         esp_ota_abort(imageProcessor->mOTAUpdateHandle);
-        imageProcessor->ReleaseBlock();
+        LogErrorOnFailure(imageProcessor->ReleaseBlock());
         PostOTAStateChangeEvent(DeviceLayer::kOtaDownloadFailed);
     }
 
@@ -381,7 +381,7 @@ void OTAImageProcessorImpl::HandleFinalize(intptr_t context)
     {
         ESP_LOGE(TAG, "esp_delta_ota_deinit() failed (%s)!", esp_err_to_name(err));
         esp_ota_abort(imageProcessor->mOTAUpdateHandle);
-        imageProcessor->ReleaseBlock();
+        LogErrorOnFailure(imageProcessor->ReleaseBlock());
         PostOTAStateChangeEvent(DeviceLayer::kOtaDownloadFailed);
     }
 

--- a/src/platform/ESP32/OTAImageProcessorImpl.cpp
+++ b/src/platform/ESP32/OTAImageProcessorImpl.cpp
@@ -187,10 +187,10 @@ esp_err_t OTAImageProcessorImpl::VerifyHeaderData(const uint8_t * buf, size_t si
             memcpy(patchHeader + headerDataRead, buf, *index);
             if (!VerifyPatchHeader(patchHeader))
             {
+                headerDataRead = 0;
                 return ESP_ERR_INVALID_VERSION;
             }
             headerDataRead      = 0;
-            *index              = PATCH_HEADER_SIZE;
             patchHeaderVerified = true;
         }
     }
@@ -264,9 +264,15 @@ esp_err_t OTAImageProcessorImpl::DeltaOTAWriteCallback(const uint8_t * buf, size
                 return ESP_ERR_INVALID_VERSION;
             }
             imageProcessor->chipIdVerified = true;
+            headerDataRead                 = 0;
 
             // Write data in headerData buffer.
-            return esp_ota_write(imageProcessor->mOTAUpdateHandle, headerData, IMG_HEADER_LEN);
+            esp_err_t err = esp_ota_write(imageProcessor->mOTAUpdateHandle, headerData, IMG_HEADER_LEN);
+            if (err != ESP_OK)
+            {
+                ESP_LOGE(TAG, "esp_ota_write failed (%s)!", esp_err_to_name(err));
+                return err;
+            }
         }
     }
 

